src/tls.c

```diff
@@ -8179,13 +8231,18 @@ static int TLSX_KeyShare_SetSupported(WOLFSSL* ssl)
     /* Delete the old key share data list. */
     extension = TLSX_Find(ssl->extensions, TLSX_KEY_SHARE);
     if (extension != NULL) {
-        TLSX_KeyShare_FreeAll((KeyShareEntry*)extension->data, ssl->heap);
+        KeyShareEntry* kse = (KeyShareEntry*)extension->data;
+    #ifdef WOLFSSL_ASYNC_CRYPT
+        if (kse && kse->lastRet == WC_PENDING_E)
+            return TLSX_KeyShare_GenKey(ssl, kse);
```

Comment:
Assuming this is completing the operation before freeing.
Can the code continue if the key generation finishes?

Reply:
No, this is completing the previous key generation done in `TLSX_KeyShare_Use`. I've fixed this to instead fall-through and call `TLSX_KeyShare_Use` again. Thanks, much cleaner.